src/shared/tpm2-util.c

```diff
@@ -2119,6 +2119,8 @@ int tpm2_create_primary(
                         /* creationData= */ NULL,
                         /* creationHash= */ NULL,
                         /* creationTicket= */ NULL);
+        if (rc == TPM2_RC_BAD_AUTH)
+                return log_debug_errno(SYNTHETIC_ERRNO(EPERM), "TPM is locked, cannot generate primary key.");
```

Comment:
Maybe `EWOULDBLOCK` or something is better as a recognizable error code? EPERM or errno which can be tracked with ERRNO_IS_NEG_PRIVILEGE() may be returned in other places.

Reply:
yeah, a recognizable error would be good here.

but EWOULDBLOCK would not be the best choice since that is just a compat alias for EAGAIN.

Maybe EDEADLK or so?